### PR TITLE
azure: various changes to the build rules

### PR DESCRIPTION
**Pull request by compnerd**
Update the build rules for the rebranch and VS changes.  We no longer modify the installed Visual Studio copy as much, add in new dependencies, checkouts, builds, update paths, update build tools, update flags for the compilers build.

### CODE REVIEW

File: .azure/vs2022.yml [hunk 1]
```diff
@@ -1302,114 +1325,153 @@ stages:
...
-               -D SQLite3_LIBRARY=$(Pipeline.Workspace)/sqlite-$(arch)-3.36.0/Library/sqlite-3.36.0/usr/lib/SQLite3.lib
-               -D SQLite3_INCLUDE_DIR=$(Pipeline.Workspace)/sqlite-$(arch)-3.36.0/Library/sqlite-3.36.0/usr/include
+              cmakeArgs:
+                -B $(Agent.BuildDirectory)/swift-asn1
+                -D BUILD_SHARED_LIBS=NO
+                -D CMAKE_BUILD_TYPE=Release
+                -D CMAKE_MT=mt
+                -D CMAKE_Swift_COMPILER=$(workspace.root)/toolchain-amd64/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr/bin/swiftc.exe
+                -D CMAKE_Swift_COMPILER_TARGET=$(platform)-unknown-windows-msvc
+                -D CMAKE_Swift_FLAGS="-resource-dir $(Pipeline.Workspace)/windows-sdk-$(arch)/Library/Developer/Platforms/Windows.platform/Developer/SDKs/Windows.sdk/usr/lib/swift -L$(Pipeline.Workspace)/windows-sdk-$(arch)/Library/Developer/Platforms/Windows.platform/Developer/SDKs/Windows.sdk/usr/lib/swift/windows"
```
Comment (tristanlabelle) [inline .azure/vs2022.yml:1335]: These don't need the `-vfsoverlay`?

**Review by tristanlabelle — APPROVED** [review]

File: .azure/vs2022.yml (hunk 1 above)
Comment (compnerd) [reply .azure/vs2022.yml:1335]: They do, but the file that I am using for injection is not available across the phases.  So for now we are modifying the VS installation.  We should start using `-sdk` with a staged SDK and avoid the modification of Visual Studio which would then not need the `-vfsoverlay` flag nor modify the installation.  cf. build.ps1

File: .azure/vs2022.yml [hunk 2]
```diff
@@ -1302,114 +1325,153 @@ stages:
...
+              cmakeArgs:
+                -B $(Agent.BuildDirectory)/swift-certificates
+                -D BUILD_SHARED_LIBS=NO
+                -D CMAKE_BUILD_TYPE=Release
+                -D CMAKE_MT=mt
+                -D CMAKE_Swift_COMPILER=$(workspace.root)/toolchain-amd64/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr/bin/swiftc.exe
+                -D CMAKE_Swift_COMPILER_TARGET=$(platform)-unknown-windows-msvc
+                -D CMAKE_Swift_FLAGS="-resource-dir $(Pipeline.Workspace)/windows-sdk-$(arch)/Library/Developer/Platforms/Windows.platform/Developer/SDKs/Windows.sdk/usr/lib/swift -L$(Pipeline.Workspace)/windows-sdk-$(arch)/Library/Developer/Platforms/Windows.platform/Developer/SDKs/Windows.sdk/usr/lib/swift/windows"
+                -G Ninja
+                -S $(Build.SourcesDirectory)/swift-package-manager
```
Comment (tristanlabelle) [inline .azure/vs2022.yml:1355]: This looks wrong. Shouldn't it be `$(Build.SourcesDirectory)/swift-certificates`?

Comment (compnerd) [reply .azure/vs2022.yml:1355]: Ugh!  Yes.

Comment (compnerd) [inline .azure/vs2022.yml:1355]:
```suggestion
                -S $(Build.SourcesDirectory)/swift-certificates
```

**Pull request closed without merging**